### PR TITLE
Integration of a Doodle-like functionality in OAE

### PR DESCRIPTION
**Pull request by Orodan**
Hi,

I would like to propose you the work we've done, Mathilde Guerin, Frederic Dooremont and myself. We integrated a Doodle-like functionality in OAE as a new type of content, to be able to schedule meetings in OAE.

We created a new module in Hilary called "oae-loodle" to handle the surveys in OAE. 
This functionality was integrated through a third application we developed at the University of La Rochelle, called "Loodle". This application is in charge of all the logic related to surveys and exposes a Rest API. The goal was to keep the module in Hilary as simple as we could.
So to be able to test my work, you should have an instance of the Loodle application running and set the correct address in OAE through the configuration page.

Loodle application : 
https://github.com/Orodan/Loodle/

To help you track the changes we've made in Hilary, we created a file in our OAE tenant at the University of La Rochelle which lists all the modified files in Hilary and the reasons of theses changes : 

https://oae.univ-lr.fr/content/ULR/VJnJlrYWb

I hope you appreciate this work and feel free to contact us if you have any questions of remarks.

Best regards

Jimmy

### CODE REVIEW

File: node_modules/oae-rest/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The `rest` module for OAE.
```
Comment (sallakarppinen) [inline node_modules/oae-rest/README.md]: The oae-rest module lives in a [separate repository](https://github.com/oaeproject/oae-rest) from Hilary so changes to it should be submitted there.

File: .idea/.name [hunk 2]
```diff
@@ -0,0 +1 @@
+Hilary
```
Comment (sallakarppinen) [inline .idea/.name]: It looks like these files sneaked past gitignore!

File: node_modules/oae-rest/README.md (hunk 1 above)
Comment (Orodan) [inline node_modules/oae-rest/README.md]: I updated gitignore to ignore the oae-rest module and I added a pull request for the oae-rest project : https://github.com/oaeproject/oae-rest/pull/29.

File: .idea/.name (hunk 2 above)
Comment (Orodan) [inline .idea/.name]: Indeed I fixed it it should be ok now.

**Comment by simong** [comment]
@Orodan it looks like you included the `node_modules` for the `oae-loodle` module. Can you exclude these?

Can you also merge this branch up with master?

**Comment by Orodan** [comment]
@simong You want me to put my forked oae in 12.3.1 that's correct ?

**Comment by sallakarppinen** [comment]
Some thoughts after running this locally:

1. The loodle option shows up in the 'Create' menu whether loodles are enabled in the tenancy's config or not - it should only be visible when the functionality is enabled;
2. Currently fails silently when a loodle cannot be created (for example, because the functionality is not enabled). Some kind of an error message should be shown when loodle creation fails;
3. When trying to make changes to a loodle, I'm getting the following error:
```
TypeError: async.forEachOf is not a function
     at Object.exports.updateVotes (/home/salla/git/oae/Hilary/node_modules/oae-loodle/lib/internal/dao/loodles.js:806:11)
     at /home/salla/git/oae/Hilary/node_modules/oae-loodle/lib/api.js:529:23
     at /home/salla/git/oae/Hilary/node_modules/oae-loodle/lib/internal/dao/users.js:24:16
     at /home/salla/git/oae/Hilary/node_modules/oae-util/lib/cassandra.js:783:16
     at onReturn (/home/salla/git/oae/Hilary/node_modules/helenus/lib/connection.js:490:7)
     at onReturn (/home/salla/git/oae/Hilary/node_modules/helenus/lib/connection.js:422:7)
     at client._reqs.(anonymous function) (/home/salla/git/oae/Hilary/node_modules/helenus-thrift/lib/thrift/connection.js:80:11)
     at Object.CassandraClient.recv_execute_cql3_query (/home/salla/git/oae/Hilary/node_modules/helenus/lib/cassandra/Cassandra.js:8343:12)
     at /home/salla/git/oae/Hilary/node_modules/helenus-thrift/lib/thrift/connection.js:83:37
     at Socket.<anonymous> (/home/salla/git/oae/Hilary/node_modules/helenus-thrift/lib/thrift/transport.js:74:11)
```
4. While the changes are not persisted due to the above error, the front-end does change to show them - this needs to be fixed as otherwise a user is likely to ignore the error message saying changes couldn't be saved.

File: node_modules/oae-loodle/lib/internal/dao/users.js [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+ */
+var _createLoodleAccount = function (ctx, user, callback) {
+
+    // Get config values
+    var host = LoodleConfig.getValue(ctx.tenant().alias, 'activity', 'host'),
+        port = LoodleConfig.getValue(ctx.tenant().alias, 'activity', 'port');
+
+    // Create the user object data
+    var data = JSON.stringify({
+        'email': user.id.replace(/:/g, '-') + '@oae.com', // We need to make sure the email is unique, even for administrator, so we use the user id
```
Comment (sallakarppinen) [inline node_modules/oae-loodle/lib/internal/dao/users.js]: This seems hacky as hell to me.

File: node_modules/oae-loodle/lib/internal/dao/users.js [hunk 4]
```diff
@@ -0,0 +1,349 @@
...
+
+    req.on('error', function (err) {
+        log().error("Can't get the loodles associated with the user : ", err.message);
+    });
+
+    req.end();
+
+};
+
+// A REMPLACER PAR UNE VERSION OU ON A JUSTE BESOIN DE FOURNIR L'ID DE L'UTILISATEUR DANS OAE
```
Comment (sallakarppinen) [inline node_modules/oae-loodle/lib/internal/dao/users.js]: Has this method been replaced with the correct version? If so, take the comment out.

File: node_modules/oae-loodle/lib/rest.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+
+// Get a named logger so we can post messages in the log
```
Comment (sallakarppinen) [inline node_modules/oae-loodle/lib/rest.js]: I don't think we need comments on why dependencies are included.

**Comment by sallakarppinen** [comment]
Also, forgot to mention in review comments, but the year in licences is wrong - should be 2016 everywhere.

**Pull request closed without merging**